pkg/cdc/sql_builder.go

```diff
@@ -0,0 +1,594 @@
+)
+
+var CDCSQLTemplates = [CDCSqlTemplateCount]struct {
+	SQL         string
+	OutputAttrs []string
+}{
+	CDCInsertTaskSqlTemplate_Idx: {
+		SQL: CDCInsertTaskSqlTemplate,
+	},
+	CDCGetTaskSqlTemplate_Idx: {
+		SQL: CDCGetTaskSqlTemplate,
+		OutputAttrs: []string{
+			"sink_uri",
+			"sink_type",
+			"sink_password",
+			"tables",
+			"filters",
+			"start_ts",
+			"end_ts",
+			"no_full",
+			"additional_config",
+		},
+	},
+	CDCShowTaskSqlTemplate_Idx: {
+		SQL: CDCShowCdcTaskSqlTemplate,
+		OutputAttrs: []string{
+			"task_id",
+			"task_name",
+			"source_uri",
+			"sink_uri",
+			"state",
+			"err_msg",
+		},
+	},
+	CDCGetTaskIdSqlTemplate_Idx: {
+		SQL:         CDCGetCdcTaskIdSqlTemplate,
+		OutputAttrs: []string{"task_id"},
+	},
+	CDCDeleteTaskSqlTemplate_Idx: {
+		SQL: CDCDeleteTaskSqlTemplate,
+	},
+	CDCUpdateTaskStateSQL_Idx: {
+		SQL: CDCUpdateTaskStateSqlTemplate,
+	},
+	CDCUpdateTaskStateAndErrMsgSQL_Idx: {
+		SQL: CDCUpdateTaskStateAndErrMsgSqlTemplate,
+	},
+	CDCInsertWatermarkSqlTemplate_Idx: {
+		SQL: CDCInsertWatermarkSqlTemplate,
+	},
+	CDCGetWatermarkSqlTemplate_Idx: {
+		SQL: CDCGetWatermarkSqlTemplate,
+		OutputAttrs: []string{
+			"db_name",
+			"table_name",
+			"watermark",
+			"err_msg",
+		},
+	},
+	CDCGetTableWatermarkSQL_Idx: {
+		SQL: CDCGetTableWatermarkSqlTemplate,
+		OutputAttrs: []string{
+			"watermark",
+		},
+	},
+	CDCUpdateWatermarkSQL_Idx: {
+		SQL: CDCUpdateWatermarkSqlTemplate,
+	},
+	CDCUpdateWatermarkErrMsgSQL_Idx: {
+		SQL: CDCUpdateWatermarkErrMsgSqlTemplate,
+	},
+	CDCDeleteWatermarkSqlTemplate_Idx: {
+		SQL: CDCDeleteWatermarkSqlTemplate,
+	},
+	CDCDeleteWatermarkByTableSqlTemplate_Idx: {
+		SQL: CDCDeleteWatermarkByTableSqlTemplate,
+	},
+	CDCGetDataKeySQL_Idx: {
+		SQL:         CDCGetDataKeySqlTemplate,
+		OutputAttrs: []string{"encrypted_key"},
+	},
+	CDCCollectTableInfoSqlTemplate_Idx: {
+		SQL: fmt.Sprintf(
+			CDCCollectTableInfoSqlTemplate,
+			catalog.SystemOrdinaryRel,
+			AddSingleQuotesJoin(catalog.SystemDatabases),
+		),
+		OutputAttrs: []string{
+			"rel_id",
+			"relname",
+			"reldatabase_id",
+			"reldatabase",
+			"rel_createsql",
+			"account_id",
+		},
+	},
+}
+
+type cdcSQLBuilder struct{}
+
+// ------------------------------------------------------------------------------------------------
+// Task SQL
+// ------------------------------------------------------------------------------------------------
+func (b cdcSQLBuilder) InsertTaskSQL(
+	accountId uint64,
+	taskId string,
+	taskName string,
+	sourceUri string,
+	sourcePwd string,
+	sinkUri string,
+	sinkTyp string,
+	sinkPwd string,
+	sinkCaPath string,
+	sinkCertPath string,
+	sinkKeyPath string,
+	tables string,
+	filters string,
+	opfilters string,
+	sourceState string,
+	sinkState string,
+	startTs string,
+	endTs string,
+	configFile string,
+	taskCreateTime time.Time,
+	state string,
+	checkpoint uint64,
+	noFull bool,
+	incrConfig string,
+	additionalConfigStr string,
+) string {
+	return fmt.Sprintf(
```

Comment:
All these Sprintf based SQL construction are subject to SQL injection.    I wonder if we can make it use param binding -- or even if not possible, values be systematically escaped here.